engines/python/setup/djl_python/streaming_utils.py

```diff
@@ -0,0 +1,179 @@
+            return StreamingUtils._transformers_neuronx_stream_generator
+        else:
+            raise ValueError(f"{engine} engine is not supported for streaming")
+
+
+    @staticmethod
+    def default_output_formatter(token_texts):
+        token_texts =  {"outputs" : token_texts}
+        json_encoded_str = json.dumps(token_texts) + "\n"
+        return bytearray(json_encoded_str.encode("utf-8"))
+
+
+    @staticmethod
+    @torch.inference_mode()
+    def _hf_model_stream_generator(model, tokenizer, inputs, **kwargs):
+        StreamingUtils._validate_inputs(inputs)
+        if not tokenizer.pad_token:
+            tokenizer.pad_token = tokenizer.eos_token
+        tokenized_inputs = tokenizer(inputs, return_tensors="pt", padding=True).to(
+                                        StreamingUtils._get_current_device())
+        input_ids = tokenized_inputs["input_ids"]
+        input_length = input_ids.shape[1]
+        all_input_ids = tokenized_inputs["input_ids"]
+        position_ids = tokenized_inputs["attention_mask"].long().cumsum(-1) - 1
+        max_new_tokens = kwargs.get("max_new_tokens", StreamingUtils.DEFAULT_MAX_NEW_TOKENS)
+        attention_mask = input_ids.new_zeros(len(inputs), input_length + max_new_tokens)
+        attention_mask[:, :input_length] = tokenized_inputs["attention_mask"]
+        past_key_values = None
+        decoding_method = StreamingUtils._get_decoding_method(**kwargs)
+        stop_generation = False
+        curr_length = input_length
+        new_tokens_count = 0
+        while True:
+            if stop_generation:
+                return
+            
+            attention_mask_curr = attention_mask[:, :curr_length]
+            outputs = model.forward(input_ids=input_ids,
+                                attention_mask=attention_mask_curr,
+                                past_key_values=past_key_values,
+                                use_cache=True
+                                )
+            next_token_ids = []
+            for logits in outputs.logits:
+                next_token_id = decoding_method(logits, all_input_ids, **kwargs)
+                next_token_ids.append(next_token_id.view(1,1))
+            token_ids = torch.cat(next_token_ids)
+            past_key_values = outputs.past_key_values
+            token_text = tokenizer.batch_decode(token_ids)
+            input_ids = token_ids.view(len(inputs), 1)
+            all_input_ids = torch.cat([all_input_ids, token_ids], dim=1)
+            attention_mask[:, curr_length] = 1
+            curr_length += 1
+            new_tokens_count += 1
+            stop_generation = StreamingUtils._has_met_stopping_criteria(token_ids, new_tokens_count,
+                                        max_new_tokens, model.config.eos_token_id)
+            yield token_text
+
+
+    @staticmethod
+    @torch.inference_mode()
+    def _transformers_neuronx_stream_generator(model, tokenizer, inputs, **kwargs):
```

Comment:
add a todo to migrate this to TNX stream solution

Reply:
is TNX implementing streaming natively in their repo and that's what we want to migrate to?